src/main/java/frc/trigon/robot/subsystems/intake/Intake.java

```diff
@@ -35,6 +35,14 @@ public IntakeConstants.IntakeState getTargetState() {
         return targetState;
     }
 
+    public boolean hasNote() {
+        return IntakeConstants.HAS_NOTE_BOOLEAN_EVENT.getAsBoolean();
+    }
+
+    public boolean isEarlyNoteCollectionDetected() {
```

Comment:
add javadocs to explain what's early note collection and what's the difference between hasNote